biodatasets/chebi_nactem/chebi_nactem.py

```diff
@@ -0,0 +1,262 @@
+                    "entities": [
+                        {
+                            "id": datasets.Value("string"),
+                            "type": datasets.Value("string"),
+                            "text": datasets.Value("string"),
+                            "offsets": datasets.Sequence([datasets.Value("int32")]),
+                        }
+                    ],
+                    "relations": [
+                        {
+                            "id": datasets.Value("string"),
+                            "type": datasets.Value("string"),
+                            "arg1": datasets.Value("string"),
+                            "arg2": datasets.Value("string"),
+                        }
+                    ],
+                }
+            )
+
+        elif self.config.schema == "bigbio_kb":
+            features = schemas.kb_features
+        else:
+            raise NotImplementedError(self.config.schema)
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download_and_extract(urls)
+
+        subset_paths = {
+            "chebi_nactem_abstr_ann1": "ChEBI/abstracts/Annotator1",
+            "chebi_nactem_abstr_ann2": "ChEBI/abstracts/Annotator2",
+            "chebi_nactem_fullpaper": "ChEBI/fullpapers",
+        }
+
+        subset_dir = os.path.join(data_dir, subset_paths[self.config.subset_id])
+
+        doc_ids = set()
+        for file_name in os.listdir(subset_dir):
+            _id = file_name.split(".")[0]
+            doc_ids.add(_id)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # Whatever you put in gen_kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "data_dir": subset_dir,
+                    "document_ids": doc_ids,
+                },
+            )
+        ]
+
+    def _generate_examples(self, data_dir: str, document_ids: Set[str]) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        label_pos_pattern = re.compile(r"(?P<entity>\S+) (?P<offsets>(;?\d+ \d+)+)")
+
+        def uid_gen():
+            _uid = 0
+            while True:
+                yield str(_uid)
+                _uid += 1
+
+        uid = iter(uid_gen())
+
+        for idx, doc_id in enumerate(document_ids):
+            doc_file = os.path.join(data_dir, doc_id + ".txt")
+            with open(doc_file) as handle:
+                text = handle.read()
+
+            ann_file = os.path.join(data_dir, doc_id + ".ann")
+
+            annotations = []
+            relations = []
+
+            with open(ann_file) as handle:
```

Comment:
Could you please switch to the functions for parsing BRAT files found in  `utils.parsing`? It's just to be 100% sure.

Reply:
Done.